data_model/src/events/data/events.rs

```diff
@@ -0,0 +1,301 @@
+//! This module contains data events
+
+use super::*;
+
+/// Trait for retrieving id from events
+pub trait IdTrait: Identifiable {
+    fn id(&self) -> &Self::Id;
+}
+
+mod asset {
+    //! This module contains `AssetEvent`, `AssetDefinitionEvent` and its impls
+
+    use super::*;
+
+    #[derive(Clone, PartialEq, Eq, Debug, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    #[non_exhaustive]
+    #[allow(missing_docs)]
+    pub enum AssetEvent {
+        Created(AssetId),
+        Deleted(AssetId),
+        Increased(AssetId),
+        Decreased(AssetId),
+        MetadataInserted(AssetId),
+        MetadataRemoved(AssetId),
+    }
+
+    impl Identifiable for AssetEvent {
+        type Id = AssetId;
+    }
+
+    impl IdTrait for AssetEvent {
+        fn id(&self) -> &AssetId {
+            match self {
+                Self::Created(id)
+                | Self::Deleted(id)
+                | Self::Increased(id)
+                | Self::Decreased(id)
+                | Self::MetadataInserted(id)
+                | Self::MetadataRemoved(id) => id,
+            }
+        }
+    }
+
+    #[derive(Clone, PartialEq, Eq, Debug, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    #[non_exhaustive]
+    #[allow(missing_docs)]
+    pub enum AssetDefinitionEvent {
+        Created(AssetDefinitionId),
+        Deleted(AssetDefinitionId),
+        MetadataInserted(AssetDefinitionId),
+        MetadataRemoved(AssetDefinitionId),
+    }
+
+    impl Identifiable for AssetDefinitionEvent {
+        type Id = AssetDefinitionId;
+    }
+
+    impl IdTrait for AssetDefinitionEvent {
+        fn id(&self) -> &AssetDefinitionId {
+            match self {
+                Self::Created(id)
+                | Self::Deleted(id)
+                | Self::MetadataInserted(id)
+                | Self::MetadataRemoved(id) => id,
+            }
+        }
+    }
+}
+
+mod peer {
+    //! This module contains `PeerEvent` and its impls
+
+    use super::*;
+
+    #[derive(Clone, PartialEq, Eq, Debug, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    #[non_exhaustive]
+    #[allow(missing_docs)]
+    pub enum PeerEvent {
+        Trusted(PeerId),
+        Untrusted(PeerId),
```

Comment:
Please rename to `Added`/`Removed`. `Trusted`/`Untrusted` are so ambiguous, I was confused by it